pkg/repositories/transformers/filters.go

```diff
@@ -86,6 +90,62 @@ func constructModelFilter(ctx context.Context, singleFilter *datacatalog.SingleP
 				JoinCondition: gormimpl.NewGormJoinCondition(sourceEntity, common.Tag),
 			}, nil
 		}
+	case *datacatalog.SinglePropertyFilter_DatasetFilter:
+		switch datasetProperty := propertyFilter.DatasetFilter.GetProperty().(type) {
+		case *datacatalog.DatasetPropertyFilter_Project:
+			project := datasetProperty.Project
+			logger.Debugf(ctx, "Constructing Dataset filter project:[%v]", project)
+			if err := validators.ValidateEmptyStringField(datasetProperty.Project, "project"); err != nil {
```

Comment:
are these empty string field validators necessary? can you have empty values for primitive one-of fields?

Reply:
Good question. If the client provides the primitive as part of the filter, then it can be misunderstood as only matching the datasets where `field=""`. Since a Datasets project, domain, name and version cannot be empty, the client should not be able to do this. So I think we should check for empty in these filters.